src/plugins/expressions/server/legacy.ts

```diff
@@ -119,9 +119,9 @@ export const createLegacyServerEndpoints = (
         clientParams: Record<string, any> = {},
         options?: any
       ) =>
-        (await core.getStartServices())[0].elasticsearch.client
-          .asScoped(request)
-          .callAsCurrentUser(endpoint, clientParams, options),
+        core.getStartServices().then(([{ elasticsearch: { client } }]) =>
```

Comment:
Perhaps, this code can be simplified if put this logic in async `onBatchItem` function.